packages/docusaurus-plugin-content-docs/src/index.ts

```diff
@@ -191,7 +197,13 @@ export default async function pluginContentDocs(
             tagsFile,
           });
         }
-        return Promise.all(docFiles.map(processVersionDoc));
+        const results = [];
+        while (docFiles.length > 0) {
+          const batch = docFiles.splice(0, Concurrency);
+          const batchResults = await Promise.all(batch.map(processVersionDoc));
+          results.push(...batchResults);
+        }
+        return results;
```

Comment:
If Git is the bottleneck, and the env variable name contains Git, then we should limit the concurrent of Git access, and not create chunks of docs processing

The variable name does not match the implementation detail.

Also chunking is not ideal because we can't start new IO work until a full chunk has ended, and we have to wait for the slowest of the 100 docs to have finished processing before starting processing the 100-200 dos chunk.

Using something like https://github.com/sindresorhus/p-queue is better